drivers/timer/nrf_rtc_timer.c

```diff
@@ -189,8 +206,160 @@ void _timer_idle_enter(s32_t sys_ticks)
 	 * immediately, meaning that we will not go to sleep.
 	 */
 	rtc_compare_set(rtc_past + (sys_ticks * RTC_TICKS_PER_SYS_TICK));
+#endif
 }
 
+
+#ifdef CONFIG_TICKLESS_KERNEL
+
+/**
+ * @brief provides total systicks programed.
+ *
+ * returns : total number of sys ticks programed.
+ */
+
+uint32_t _get_program_time(void)
+{
+	return expected_sys_ticks;
+}
+
+/**
+ * @brief provides total systicks remaining since last programing of RTC.
+ *
+ * returns : total number of sys ticks remaining since last RTC programing.
+ */
+
+uint32_t _get_remaining_program_time(void)
+{
+	u64_t rtc_elapsed;
+	u32_t rtc_now;
+
+	if (!expected_sys_ticks) {
+		return 0;
+	}
+
+	rtc_now = RTC_COUNTER;
+
+	/* Number of RTC cycles passed since last RTC Programing*/
+	rtc_elapsed = (rtc_now - rtc_past) & RTC_MASK;
+
+	return (expected_sys_ticks - (rtc_elapsed / RTC_TICKS_PER_SYS_TICK));
```

Comment:
reuse _get_elapsed_program_time instead of duplicating code for rtc_elapsed calculation.